fix(Adaptor): Lambda Response Streaming With Content Type

watany-dev opened the pull request:
fixed: https://github.com/honojs/hono/issues/1672

I'd like to share that we have completed actual device verification with AWS Lambda for the magic method awslambda.HttpResponseStream.from, which adds metadata (statusCode, Headers) to the responseStream.

### Author should do the followings, if applicable

- [x] Add tests
- [x] Run tests
- [ ] `yarn denoify` to generate files for Deno

yusukebe commented:
Hi @watany-dev !

I didn't know `application/vnd.awslambda.http-integration-response`!

Like the CI is failing, you can't write the actual logic in a `.d.ts` file. I think you should write `streamifyResponse` and `awslambda.HttpResponseStream.from` in `handler.ts` or create a new file for them.

watany-dev commented:
@yusukebe 
These specific AWS Lambda headers and methods are already executed within the AWS environment, so they are not necessary in the .d.ts file, right? I have removed them and added a reference to the source library in the comments. (This is the official sample implementation for when users are setting up the AWS Lambda environment.)

usualoma commented:
FYI, in my environment, this PR code fixed the problem and `text/event-stream` , and `text/html; charset=UTF-8` are now sent.
@watany-dev @yusukebe

yusukebe commented:
Ah, this PR version works. My configuration was wrong; the content-types are now correct.

One thing not directly related to this PR: HTTP Streaming with `Suspense` and `renderToReadableStream()` does not work.

watany-dev commented:
@usualoma @yusukebe 
Thanks for confirming this.

I can't find a track record of Suspense or renderToReadableStream working with AWS lambda, even with the original React, so it may be too complicated to handle here.

yusukebe commented:
Hmm.

This PR is OK, so I'll merge it later.

However, I think we might make `Suspense` and `renderToReadableStream()` work on Lambda function URLs. I saw an example here that suggests it may be possible:
https://github.com/aws-samples/serverless-patterns/tree/main/lambda-streaming-ttfb-write-sam

watany-dev commented:
Thanks, this is interesting and I'll check the behavior, I hope it's a bug in adaptor, otherwise it's tricky...

yusukebe commented:
Anyway, I'm merging it now. Thanks!

yusukebe commented:
Created the issue #1678.
